store/sqlstore/shared_channel_store.go

```diff
@@ -581,28 +587,112 @@ func (s SqlSharedChannelStore) GetUser(userID string, channelID string, remoteID
 	return &scu, nil
 }
 
-// UpdateUserLastSyncAt updates the LastSyncAt timestamp for the specified SharedChannelUser.
-func (s SqlSharedChannelStore) UpdateUserLastSyncAt(id string, syncTime int64) error {
+// GetUser fetches all shared channel user records based on userID.
+func (s SqlSharedChannelStore) GetUser(userID string) ([]*model.SharedChannelUser, error) {
```

Comment:
nit: naming is a bit confusing here, as I would only expect one user to be returned. Maybe we could change this to `GetSharedUsersForUser`? 1/5